dagshub/data_engine/model/query_result.py

```diff
@@ -623,22 +694,63 @@ def visualize(self, visualizer: Literal["dagshub", "fiftyone"] = "dagshub", **kw
             return sess
 
     def annotate(
-        self, open_project=True, ignore_warning=True, fields_to_embed=None, fields_to_exclude=None
+        self,
+        open_project=True,
+        ignore_warning=True,
+        fields_to_embed=None,
+        fields_to_exclude=None,
+        model_name=None,
+        model_version="latest",
+        model_repo=None,
+        model_pre_hook=lambda x: x,
+        model_post_hook=None,
+        model_batch_size=1,
+        model_metadata_column="annotation",
+        use_remote_backend=False,
     ) -> Optional[str]:
         """
         Sends all the datapoints returned in this QueryResult to be annotated in Label Studio on DagsHub.
+        Alternatively, uses MLFlow to automatically label datapoints.
```

Comment:
1) Split this into a different function, something like `annotate_with_mlflow`. I understand that this name of a function is kind of too generic, but making it have two wildly different behaviors depending on args is imo a bad pattern.
2) Add typehints to the args (you can add them to this one too if you want)
3) When you move it to a separate function, you can probably drop all the `model_` prefixes

Reply:
Chat with Dean about a good name for this function. `get_predictions` is also very generic and might conflict with what I'm doing right now with the annotation conversions.
Probably needs a `from/with_mlflow_model` or smth like that